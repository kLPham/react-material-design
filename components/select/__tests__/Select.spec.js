import React from 'react';
import { shallow } from 'enzyme';
import Select from '../index';
import Menu from '../../menu';
import MenuItem from '../../menu/menuItem';

describe('<Select />', () => {
    it('should render', () => {
        const select = shallow(
            <Select>
                <Menu selectMenu>
                    <MenuItem value="Bread, Cereal, Rice, and Pasta" />
                    <MenuItem value="Vegetables" />
                    <MenuItem value="Fruit" />
                    <MenuItem value="Milk, Yogurt, and Cheese" />
                    <MenuItem value="Meat, Poultry, Fish, Dry Beans, Eggs, and Nuts" />
                    <MenuItem value="Fats, Oils, and Sweets" />
                </Menu>
            </Select>,
    );
        expect(select).toMatchSnapshot();
    });
    it('should render disabled', () => {
        const disable = true;
        const select = shallow(
            <Select disable={disable}>
                <Menu selectMenu>
                    <MenuItem value="Bread, Cereal, Rice, and Pasta" />
                    <MenuItem value="Vegetables" />
                    <MenuItem value="Fruit" />
                    <MenuItem value="Milk, Yogurt, and Cheese" />
                    <MenuItem value="Meat, Poultry, Fish, Dry Beans, Eggs, and Nuts" />
                    <MenuItem value="Fats, Oils, and Sweets" />
                </Menu>
            </Select>,
      );
        expect(select).toMatchSnapshot();
    });
});
