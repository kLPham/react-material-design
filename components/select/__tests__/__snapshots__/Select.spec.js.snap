// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Select /> should render 1`] = `
<div
  className="mdc-select"
  role="listbox"
  tabIndex="0"
>
  <span
    className="mdc-select__selected-text"
  >
    Pick a food group
  </span>
  <div
    className="mdc-simple-menu mdc-select__menu"
  >
    <ul
      className="mdc-list mdc-simple-menu__items"
    >
      <li
        aria-disabled="true"
        className="mdc-list-item"
        id="grains"
        role="option"
      >
        Pick a food group
      </li>
      <li
        className="mdc-list-item"
        id="grains"
        role="option"
        tabIndex="0"
      >
        Bread, Cereal, Rice, and Pasta
      </li>
      <li
        className="mdc-list-item"
        id="vegetables"
        role="option"
        tabIndex="0"
      >
        Vegetables
      </li>
      <li
        className="mdc-list-item"
        id="fruit"
        role="option"
        tabIndex="0"
      >
        Fruit
      </li>
      <li
        className="mdc-list-item"
        id="dairy"
        role="option"
        tabIndex="0"
      >
        Milk, Yogurt, and Cheese
      </li>
      <li
        className="mdc-list-item"
        id="meat"
        role="option"
        tabIndex="0"
      >
        Meat, Poultry, Fish, Dry Beans, Eggs, and Nuts
      </li>
      <li
        className="mdc-list-item"
        id="fats"
        role="option"
        tabIndex="0"
      >
        Fats, Oils, and Sweets
      </li>
    </ul>
  </div>
</div>
`;

exports[`<Select /> should render disabled 1`] = `
<div
  className="mdc-select"
  role="listbox"
  tabIndex="0"
>
  <span
    className="mdc-select__selected-text"
  >
    Pick a food group
  </span>
  <div
    className="mdc-simple-menu mdc-select__menu"
  >
    <ul
      className="mdc-list mdc-simple-menu__items"
    >
      <li
        aria-disabled="true"
        className="mdc-list-item"
        id="grains"
        role="option"
      >
        Pick a food group
      </li>
      <li
        className="mdc-list-item"
        id="grains"
        role="option"
        tabIndex="0"
      >
        Bread, Cereal, Rice, and Pasta
      </li>
      <li
        className="mdc-list-item"
        id="vegetables"
        role="option"
        tabIndex="0"
      >
        Vegetables
      </li>
      <li
        className="mdc-list-item"
        id="fruit"
        role="option"
        tabIndex="0"
      >
        Fruit
      </li>
      <li
        className="mdc-list-item"
        id="dairy"
        role="option"
        tabIndex="0"
      >
        Milk, Yogurt, and Cheese
      </li>
      <li
        className="mdc-list-item"
        id="meat"
        role="option"
        tabIndex="0"
      >
        Meat, Poultry, Fish, Dry Beans, Eggs, and Nuts
      </li>
      <li
        className="mdc-list-item"
        id="fats"
        role="option"
        tabIndex="0"
      >
        Fats, Oils, and Sweets
      </li>
    </ul>
  </div>
</div>
`;
